app/src/main/java/com/vishnu/whatsappcleaner/MainViewModel.kt

```diff
@@ -43,17 +44,32 @@ class MainViewModel(private val application: Application) : AndroidViewModel(app
 
     private val storeData = StoreData(application.applicationContext)
 
+    private val _isGridView = MutableStateFlow(false)
+    val isGridView: StateFlow<Boolean> = _isGridView.asStateFlow()
+
     private val _directoryItem =
         MutableStateFlow<ViewState<Pair<String, List<ListDirectory>>>>(ViewState.Loading)
     val directoryItem: StateFlow<ViewState<Pair<String, List<ListDirectory>>>> =
         _directoryItem.asStateFlow()
 
     init {
         viewModelScope.launch(Dispatchers.Default) {
+            storeData.isGridViewFlow.collect {
+                _isGridView.value = it
+            }
             getDirectoryList()
         }
     }
 
+    fun toggleViewType() {
+        viewModelScope.launch {
+            val current = storeData.isGridViewFlow.first()
+            val toggled = !current
+            // Save the current view type to Store
```

Comment:
no need of this comment, IMO.

Reply:
@VishnuSanal Thank you for the suggestion. I will remove it.